server/src/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1078,6 +1082,16 @@ public void cleanupStorage(boolean recurring) {
                         }
                     }
 
+                    //destroy snapshots in destroying state in snapshot_store_ref
+                    List<SnapshotDataStoreVO>  ssSnapshots = _snapshotStoreDao.listByState(ObjectInDataStoreStateMachine.State.Destroying);
+                    for(SnapshotDataStoreVO ssSnapshotVO : ssSnapshots){
+                        try {
+                            _snapshotService.deleteSnapshot(snapshotFactory.getSnapshot(ssSnapshotVO.getSnapshotId(), DataStoreRole.Image));
```

Comment:
@koushik-das If there is any failure in deleteSnapshot() then cleanup will be retried. DB entry for such snapshots will only be removed on successful deletion. Destroying check  introduced in SnapshotObject class is to make sure that DB entry doesn't gets deleted.